### PR TITLE
CRUD Implementation

### PR DESCRIPTION
**Pull request by Gen0pp**

### CODE REVIEW

File: database/config.js [hunk 1]
```diff
@@ -52,3 +57,63 @@ module.exports.getGame = (game, cb) => {
     }
   });
 };
+
+// UPDATE
+
+const updateGame = (game, updateInfo, cb) => {
+  console.log('updateGame()');
+  Game.updateOne({ proxyId: game.proxyId }, updateInfo, { upsert: true }, (err, data) => {
```
Comment (ingridclaire) [inline database/config.js:65]: what is the upsert: true property here?

File: database/config.js [hunk 2]
```diff
@@ -52,3 +57,63 @@ module.exports.getGame = (game, cb) => {
...
+   }
+)
+*/
+
+
+// DELETE ONE
+
+const deleteSingle = (gameID, cb) => {
+  console.log('deleteSingle() ' + gameID);
+  Game.findOneAndDelete({ proxyId: gameID }, (error, query) => {
```
Comment (ingridclaire) [inline database/config.js:92]: do you know what the different is between this findOneAndDelete and deleteOne? I used deleteOne. Thoughts?

File: server/index.js [hunk 3]
```diff
@@ -20,4 +21,65 @@ app.get('/media', (req, res) => {
   db.getGame(game, callback);
 });
 
+// POST NEW GAME
+app.post('/game', (req, res) => {
+  const game = req.body;
+  const callback = (err, data) => {
+    if (err) {
+      res.status(404).json(err);
```
Comment (ingridclaire) [inline server/index.js:29]: would status 500 be better to indicate it was a problem on the backend? I'm still trying to figure out when to use 500.

File: server/index.js [hunk 4]
```diff
@@ -20,4 +21,65 @@ app.get('/media', (req, res) => {
...
+    } else {
+      res.status(200).json(data);
+    }
+  };
+  db.updateGame(gameID, game, callback);
+});
+
+// DELETE SINGLE GAME
+app.post('/delete/:id', (req, res) => {
```
Comment (ingridclaire) [inline server/index.js:52]: why a post method for deleting a game? should it be app.delete?

File: server/index.js [hunk 5]
```diff
@@ -20,4 +21,65 @@ app.get('/media', (req, res) => {
...
+      res.status(404).json(err);
+    } else {
+      res.status(200).json(data);
+    }
+  };
+  db.deleteSingle(game, callback);
+});
+
+// DELETE ALL GAMES
+app.post('/delete/all', (req, res) => {
```
Comment (ingridclaire) [inline server/index.js:66]: delete instead of post again?

File: README.md [hunk 6]
```diff
@@ -1 +1,38 @@
...
+
+####API
+
+|REST|API ENDPOINTS|
+|--:|:-:|
+CREATE|/game/
+READ|/media/
+UPDATE|/game/:id
+DELETE|/delete/:id
+DELETE(ALL)|/delete/all
```
Comment (ingridclaire) [inline README.md:38]: you could consider giving a little more detail here, what specifically will you get back, and what kind of data should be in body? not sure if there's a standard..

File: database/config.js (hunk 2 above)
Comment (Gen0pp) [reply database/config.js:92]: They are the same except findOneAnDelete accepts a sort parameter. https://docs.mongodb.com/manual/reference/method/db.collection.findOneAndDelete/

File: server/index.js (hunk 4 above)
Comment (Gen0pp) [reply server/index.js:52]: Yes it should be changed to delete, they do work the same right now

File: database/config.js (hunk 1 above)
Comment (Gen0pp) [reply database/config.js:65]: It adds the document if it does not exist

File: server/index.js (hunk 3 above)
Comment (Gen0pp) [reply server/index.js:29]: Not too sure I need to look up the proper status codes

**Pull request merged**